apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,367 @@
+    var value = this['_' + profile + 'Connected'];
+    if (value !== connected) {
+      this['_' + profile + 'Connected'] = connected;
+
+      // Raise an event for the profile connection changes.
+      window.dispatchEvent(new CustomEvent('bluetoothprofileconnectionchange',
+      {
+        name: profile,
+        connected: connected
+      }));
+      // if (profile === 'opp' && this.transferIcon) {
+      //   this.transferIcon.update();
+      // }
+      // if (profile === 'a2dp' && this.headphoneIcon) {
+      //   this.headphoneIcon.update();
+      // }
+    }
+  },
+
+  /**
+   * Check if bluetooth profile is connected.
+   *
+   * @public
+   * @param {String} profile profile name
+   * @return {Boolean} connected state
+   */
+  isProfileConnected: function bt_isProfileConnected(profile) {
+    var isConnected = this['_' + profile + 'Connected'];
+    if (isConnected === undefined) {
+      return false;
+    } else {
+      return isConnected;
+    }
+  },
+
+  /**
+   * Initialize bluetooth module.
+   *
+   * @public
+   */
+  start: function bt_start() {
+    if (!window.navigator.mozSettings || !window.navigator.mozBluetooth) {
+      return;
+    }
+
+    this._bluetooth = window.navigator.mozBluetooth;
+    SettingsListener.observe('bluetooth.enabled', true, function(value) {
+      if (!this._bluetooth) {
+        // roll back the setting value to notify the UIs
+        // that Bluetooth interface is not available
+        if (value) {
+          SettingsListener.getSettingsLock().set({
+            'bluetooth.enabled': false
+          });
+        }
+        return;
+      }
+    }.bind(this));
+
+    this._attributeHandler = this._bluetoothAttrChangeHandler.bind(this);
+
+    // watch attributechanged event
+    this._bluetooth.addEventListener('attributechanged',
+      this._attributeHandler);
+
+    // clear defaultAdapter and listener once adapter is removed
+    this._bluetooth.addEventListener('adapterremoved', () => {
+      this._adapter = null;
+      this._bluetooth.removeEventListener(this._attributeHandler);
+    });
+
+    // if bluetooth is enabled in booting time, try to get adapter now
+    this.debug('init bluetooth adapter');
+    this._adapter = this._bluetooth.defaultAdapter;
+    this._adapter.addEventListener('attributechanged',
+      this._bluetoothAdapterAttrChangeHandler.bind(this));
+
+    /* In file transfering case:
+     * since System Message can't be listened in two js files within a app,
+     * so we listen here but dispatch events to bluetooth_transfer.js
+     * while getting bluetooth file transfer start/complete system messages
+     */
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-start',
+      this._relayMessageEvent.bind(this, 'bluetooth-opp-transfer-start')
+    );
+
+    navigator.mozSetMessageHandler('bluetooth-opp-transfer-complete',
+      this._relayMessageEvent.bind(this, 'bluetooth-opp-transfer-complete')
```

Comment:
Since you wanted to relay the message via function `_relayMessageEvent`, it will do `this._setProfileConnected(this.Profiles.OPP, true);`. There is no case for `this._setProfileConnected(this.Profiles.OPP, false);`. Will you still to combine them? Or checking the message name before set profile connected `true/false`.